opm/autodiff/SimulatorFullyImplicitBlackoilEbos.hpp

```diff
@@ -146,11 +147,21 @@ class SimulatorFullyImplicitBlackoilEbos
         extractLegacyPoreVolume_();
         extractLegacyDepth_();
 
+        // communicate the initial solution to ebos
+        if (timer.initialStep()) {
+            convertInput(/*iterationIdx=*/0, state, ebosSimulator_ );
+            ebosSimulator_.model().invalidateIntensiveQuantitiesCache(/*timeIdx=*/0);
+        }
+
         if (output_writer_.isRestart()) {
             // This is a restart, populate WellState and ReservoirState state objects from restart file
+            ReservoirState state(grid().size(0), grid().numFaces(), phaseUsage_.num_phases);
```

Comment:
Why create this object and not use the already existing (but now shadowed) `state`? If you really want a new one, the one used inside this scope should have a different name.